### PR TITLE
PF4: Five Pathfinders cards.

### PR DESCRIPTION
**Pull request by kberg**
Depends on PF3

### CODE REVIEW

File: src/cards/pathfinders/NewVenice.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+          }).nbsp.tile(TileType.OCEAN_CITY, false, true);
+        }),
+        description: 'Requires 3 ocean tiles. Lose 2 plants. Increase your energy production 1 step and your M€ production 2 steps. ' +
+            'Place this tile on top of an existing ocean tile, IGNORING NORMAL PLACEMENT RESTRICTIONS FOR CITIES. The tile counts as a city as well as an ocean.',
+      },
+    });
+  }
+
+  public canPlay(player: Player): boolean {
+    return super.canPlay(player) && (player.plants > 2);
```
Comment (bafolts) [inline src/cards/pathfinders/NewVenice.ts:39]: Should this be `player.plants >= 2` ? The requirement only seems to be that 2 plants are lost. Do we need to check for 3 oceans?

File: src/cards/pathfinders/RareEarthElements.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+          // TODO(kberg): add new tile image.
+          b.production(((pb) => pb.megacredits(1))).slash().emptyTile().text('special');
+        }),
+        description: 'Increase your M€ production by 1 for every special tile you own on Mars.',
+      },
+    });
+  }
+
+
+  private static INVALID_TILES = [TileType.GREENERY, TileType.OCEAN, TileType.CITY];
```
Comment (bafolts) [inline src/cards/pathfinders/RareEarthElements.ts:31]: I would use `Set` here as that is how the data is being used.

**Review by bafolts — APPROVED** [review]

File: src/cards/pathfinders/NewVenice.ts (hunk 1 above)
Comment (kberg) [reply src/cards/pathfinders/NewVenice.ts:39]: Good catch. Also oceans are checked in super.canPlay().

File: src/cards/pathfinders/RareEarthElements.ts (hunk 2 above)
Comment (kberg) [reply src/cards/pathfinders/RareEarthElements.ts]: Seems like a micro-optimization given how little this will impact overall performance, but done.

**Pull request merged**